src/main/scala/is/hail/table/Table.scala

```diff
@@ -1138,4 +1139,95 @@ class Table(val hc: HailContext, val tir: TableIR) {
     assert(hintPartitioner.forall(p => p.pkType.types.sameElements(orderedKTType.pkType.types)))
     OrderedRVD.coerce(orderedKTType, rvd, None, hintPartitioner)
   }
+
+  def intervalJoin(other: Table, fieldName: String): Table = {
+    assert(other.keySignature.exists(s => s.size == 1 && s.types(0).isInstanceOf[TInterval]))
+    val intervalType = other.keySignature.get.types(0).asInstanceOf[TInterval]
+    assert(keySignature.exists(s => s.size == 1 && s.types(0) == intervalType.pointType))
+
+    val leftORVD = rvd match {
+      case ordered: OrderedRVD => ordered
+      case unordered =>
+        OrderedRVD.coerce(
+          new OrderedRVDType(key.get.toArray, key.get.toArray, signature),
+          unordered)
+    }
+
+
+    val typOrdering = intervalType.pointType.ordering
+
+    val typToInsert: Type = other.valueSignature
+
+    val (newRVType, ins) = signature.unsafeStructInsert(typToInsert, List(fieldName))
+
+    val partBc = hc.sc.broadcast(leftORVD.partitioner)
+    val rightSignature = other.signature
+    val rightKeyFieldIdx = other.keyFieldIdx.get(0)
+    val rightValueFieldIdx = other.valueFieldIdx
+    val partitionKeyedIntervals = other.rvd.boundary.crdd
+      .cflatMap { (ctx, rv) =>
```

Comment:
You can get away with just `flatMap` here, you don't seem to use the context or the context's region at all.